includes/data-port/class-sensei-data-port-job.php

```diff
@@ -231,6 +247,31 @@ public function get_status() {
 	 */
 	abstract public function get_tasks();
 
+	/**
+	 * Get the configuration for expected files.
+	 *
+	 * @return array {
+	 *    @type array $$component {
+	 *        @type callable $validator  Callback to handle validating the file before save (optional).
+	 *        @type array    $mime_types Expected mime-types for the file.
+	 *    }
+	 * }
+	 */
+	abstract public static function get_file_config();
+
+	/**
+	 * Initialize and restore state of task.
+	 *
+	 * @param string $task_class Class name of task class.
+	 *
+	 * @return Sensei_Data_Port_Task_Interface
+	 */
+	protected function initialize_task( $task_class ) {
```

Comment:
An alternative to this would be to have the sub-class provide an array of class names and this parent class handles setting `$tasks`.